src/common/events/events_io.cpp

```diff
@@ -105,16 +105,20 @@ ReaderFromAsyncReader::ReaderFromAsyncReader(EventLoop &event_loop, mio::AsyncRe
 mio::ExpectedSize ReaderFromAsyncReader::Read(
 	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end) {
 	mio::ExpectedSize read;
-	error::Error err;
 	bool finished = false;
-	err = reader_->AsyncRead(start, end, [this, &finished, &read](mio::ExpectedSize num_read) {
-		read = num_read;
-		finished = true;
-		event_loop_.Stop();
+	event_loop_.Post([start, end, this, &finished, &read]() {
```

Comment:
Wonder if we have more of these lying around.

Reply:
Yeah me too. I know we call the handler directly in some places, but it seems to be working fine. This is just a particular place where it wasn't working, but the event loop trickery is a bit of a special case.